system/include/emscripten/em_js.h

```diff
@@ -79,3 +79,32 @@
 
 #define EM_ASYNC_JS(ret, name, params, ...) _EM_JS(ret, name, __asyncjs__##name, params,          \
   "{ return Asyncify.handleAsync(async () => " #__VA_ARGS__ "); }")
+
+/*
+ * EM_JS_DEPS: Use this macro to declare indirect dependencies on JS symbols.
+ * The first argument is just unique name for the set of dependencies.  The
+ * second argument is a C string that lists JS library symbols in the same way
+ * they would be specified in the DEFAULT_LIBRARY_FUNCS_TO_INCLUDE command line
+ * setting.
+ *
+ * For example, if your code contains an EM_ASM or EM_JS block that make use of
+ * the allocate and stackSave JS library functions then you might write this in
+ * your library source code:
+ *
+ *   EM_JS_DEPS(mylib_dep, "$allocate,$stackSave");
+ *
+ * The emscripten linker will then pick this up and make sure those symbols get
+ * included in the JS support library.
+ *
+ * Dependencies declared in this way will be included if-and-only-if the object
+ * file (translation unit) in which they exist is included by the linker, so
+ * it makes sense co-locate them with the EM_JS or EM_ASM code they correspond
+ * to.
+ */
```

Comment:
The other large comment in this file uses C++-style comments.  Can we be consistent.